docs/proposals/container-runtime-interface/capabilities.md

```diff
@@ -0,0 +1,221 @@
+        // In addition to the capabilities listed here, the pod author can add additional capabilities
+        // specified in the AllowedCapabilities field.
+        DefaultCapabilities []api.Capability `json:"defaultCapabilities,omitempty"`
+...
+}
+```
+
+The `Capabilities` struct in `pkg/api/types.go`, and `pkg/api/v1/types.go` gets a new `RequestedSet` field:
+
+```
+// Capabilities represent POSIX capabilities that can be added or removed to a running container to/from a default set.
+type Capabilities struct {
+        // Capabilities added to the default capabilities
+        Add []Capability `json:"add,omitempty"`
+        // Capabilities removed from the default capabilities
+        Drop []Capability `json:"drop,omitempty"`
+        // Requested set of capabilities
+        RequestedSet []Capability `json:"requestedSet,omitempty"`
+}
+```
+
+The `Capability` struct in `pkg/kubelet/api/v1alpha1/runtime/api.proto` replaces the existing `add_capabilities` and `drop_capabilities` fields with a new `requested_capabilities` field:
+
+```
+message Capability {
+    // List of capabilities to retain.
+    repeated string requested_capabilities = 1;
+}
+```
+
+## Use Cases
+
+### User specifies `requestedSet` only in the pod spec:
+
+The simplest use case is when a user only sets the `requestedSet` field in a new pod spec:
+
+```
+  spec:
+    containers:
+      securityContext:
+        capabilities:
+          requestedSet:
+          - MKNOD, NET_ADMIN, SETPCAP, AUDIT_WRITE, ...
+```
+
+The resolved requested set equals `requestedSet`.
+
+### User specifies `add`/`drop` but no `requestedSet` in the pod spec:
+
+The second primary use case will be dealing with backwards-compatibility with existing pod specs that may specify `add` or `remove`:
+
+```
+  spec:
+    containers:
+      securityContext:
+        capabilities:
+          add:
+          - NET_ADMIN
+          drop:
+          - MKNOD
+```
+
+The resolved requested set is calculated in the API server based on a _default set_.
+
+The _default set_ is determined as follows:
+
+- The API server has a hard-coded set of capabilities (for illustrative purposes, called `HardcodedDefaultCapabilities`)
+- If admission control is not enabled, or admission control is enabled but there are no capabilities-related fields set on the `PodSecurityPolicySpec`, then `HardcodedDefaultCapabilities` is used as the default set.
+- Else (i.e. if admission control is enabled and the `PodSecurityPolicySpec` (PSP) has any capabilities-related fields set), then:
+   - If the `DefaultCapabilities` field is set, that is merged with any `DefaultAddCapabilities` and `RequiredDropCapabilities` set on the same `PodSecurityPolicySpec`.
+   - Else (i.e. the `DefaultCapabilities` field is not set), then:
+       - If the submitted pod's `SecurityContext` has a `requested-set` field, that is merged with any `DefaultAddCapabilities` and `RequiredDropCapabilities` set on the same `PodSecurityPolicySpec`.
```

Comment:
Actually I'll remove the whole line since it contradicts with https://github.com/kubernetes/kubernetes/pull/35910#discussion_r91981066.